Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp

```diff
@@ -859,19 +859,28 @@ void MediaPlayerPrivateGStreamerMSE::markEndOfStream(MediaSourcePrivate::EndOfSt
     updateStates();
 }
 
+void MediaPlayerPrivateGStreamerMSE::unmarkEndOfStream()
+{
+    GST_DEBUG("Unmarking end of stream");
+    m_eosPending = false;
+}
+
 MediaTime MediaPlayerPrivateGStreamerMSE::currentMediaTime() const
 {
+    float cachedPosition = m_cachedPosition;
     MediaTime position = MediaPlayerPrivateGStreamer::currentMediaTime();
+    float playbackProgress = std::abs(position.toFloat() - cachedPosition);
 
-    if (m_eosPending && (paused() || (position >= durationMediaTime()))) {
```

Comment:
I think the "position is fetched from video decoder" sentence doesn't apply to all the platforms. Raspberry asks to the pipeline and not directly to the video decoder: https://github.com/WebPlatformForEmbedded/WPEWebKit/blob/master/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp#L388